jvb/src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -365,6 +365,16 @@ public Endpoint(
         this(id, conference, parentLogger, iceControlling, Clock.systemUTC());
     }
 
+    /**
+     * Gets the endpoints in the conference in LastN order, with this {@link Endpoint} removed.
+     */
+    private List<AbstractEndpoint> getOrderedEndpoints()
+    {
+        List<AbstractEndpoint> allOrderedEndpoints = new LinkedList<>(getConference().getOrderedEndpoints());
+        allOrderedEndpoints.remove(this);
```

Comment:
I didn't think about it much (my main goal was to simplify the flow, so I can apply my "follow dominant speaker" changes easier), but funnily enough this seems to perform better than streams. I used the folloing to test, let me know if you have other ideas to test (otherwise I'll just change it to an ArrayList, I don't think we do other operations on it that would make a linked list faster):

```
class BitrateControllerPerfTest : StringSpec() {
    private val random = Random(1234)

    val shortList = listOf("A", "B", "C", "D")
    val longerList = listOf("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P")
    val times = 20_000_000
    init {
        "Test".config(enabled = true) {
            for (list in listOf(listOf("Warm", "u", "p"), shortList, longerList)) {
                val r1 = mutableListOf<Duration>()
                val r2 = mutableListOf<Duration>()
                val r3 = mutableListOf<Duration>()
                val r4 = mutableListOf<Duration>()
                val r5 = mutableListOf<Duration>()
                val r6 = mutableListOf<Duration>()
                repeat(5) {
                    r1 += run(list) { l, e -> o1(l, e) }
                    r2 += run(list) { l, e -> o2(l, e) }
                    r3 += run(list) { l, e -> o3(l, e) }
                    r4 += run(list) { l, e -> o4(l, e) }
                    r5 += run(list) { l, e -> o5(l, e) }
                    r6 += run(list) { l, e -> o6(l, e) }
                }
                println("With list=$list")
                println("o1: ${r1.avg()}        $r1")
                println("o2: ${r2.avg()}        $r2")
                println("o3: ${r3.avg()}        $r3")
                println("o4: ${r4.avg()}        $r4")
                println("o5: ${r5.avg()}        $r5")
                println("o6: ${r6.avg()}        $r6")
                println("---")
            }
        }
    }

    private fun List<Duration>.avg() = Duration.ofNanos(map { it.toNanos() }.sum() / size)

    private fun run(list: List<String>, f: (List<String>, String) -> List<String>): Duration {
        val start = System.nanoTime()
        repeat(times) {
            f(list, list[abs(random.nextInt()) % list.size])
        }
        val stop = System.nanoTime()
        return (stop - start).nanos
    }

    private fun o1(l: List<String>, e: String): List<String> {
        val res = LinkedList(l)
        res.remove(e)
        return res
    }

    private fun o2(l: List<String>, e: String): List<String> {
        return l.stream().filter { it != e }.toList()
    }

    private fun o3(l: List<String>, e: String): List<String> {
        val res = ArrayList<String>(l.size)
        l.forEach {
            if (it != e) {
                res.add(it)
            }
        }
        return res
    }

    private fun o4(l: List<String>, e: String): List<String> {
        val res = LinkedList<String>()
        l.forEach {
            if (it != e) {
                res.add(it)
            }
        }
        return res
    }

    private fun o5(l: List<String>, e: String): List<String> {
        return l.stream().filter { it != e }.collect(Collectors.toList())
    }

    private fun o6(l: List<String>, e: String): List<String> {
        val res = ArrayList(l)
        res.remove(e)
        return res
    }
}

```



The results:
```
With list=[A, B, C, D]
o1: PT0.946522705S        [PT1.12175796S, PT0.864302441S, PT0.88258944S, PT0.992391484S, PT0.871572201S]
o2: PT2.527977189S        [PT3.338222391S, PT2.199832695S, PT2.420955155S, PT2.439432182S, PT2.241443526S]
o3: PT1.238375498S        [PT1.310641124S, PT1.174358349S, PT1.25451887S, PT1.272078685S, PT1.180280465S]
o4: PT1.059412331S        [PT1.085235538S, PT1.029784163S, PT1.09391613S, PT1.078398377S, PT1.009727449S]
o5: PT2.306902876S        [PT2.275407939S, PT2.425750047S, PT2.364697223S, PT2.302489618S, PT2.166169555S]
o6: PT0.924653748S        [PT0.920152692S, PT0.880809085S, PT0.966745151S, PT0.940011884S, PT0.91554993S]
---
With list=[A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P]
o1: PT2.339013507S        [PT2.254260406S, PT2.203495566S, PT2.212700089S, PT2.34359944S, PT2.681012037S]
o2: PT5.172946717S        [PT4.905352347S, PT5.073093654S, PT5.210852271S, PT4.969328078S, PT5.706107238S]
o3: PT3.308663411S        [PT3.231956547S, PT3.418924182S, PT3.450413149S, PT3.223454725S, PT3.218568455S]
o4: PT3.178143994S        [PT3.240716589S, PT3.165990234S, PT3.223012512S, PT3.191998367S, PT3.069002271S]
o5: PT5.70544231S        [PT5.149150118S, PT5.163969926S, PT5.098700432S, PT6.787973047S, PT6.327418029S]
o6: PT1.643821372S        [PT1.481260213S, PT1.505072266S, PT1.44297631S, PT1.770460351S, PT2.019337722S]
```

Reply:
Huh!  Go figure.